pvlib/iotools/bsrn.py

```diff
@@ -0,0 +1,146 @@
+
+    =======================  ======  ==========================================
+    Key                      Format  Description
+    =======================  ======  ==========================================
+    day                      int     Day of the month 1-31
+    minute                   int     Minute of the day 0-1439
+    ghi                      float   Mean global horizontal irradiance [W/m^2]
+    ghi_std                  float   Std. global horizontal irradiance [W/m^2]
+    ghi_min                  float   Min. global horizontal irradiance [W/m^2]
+    ghi_max                  float   Max. global horizontal irradiance [W/m^2]
+    dni                      float   Mean direct normal irradiance [W/m^2]
+    dni_std                  float   Std. direct normal irradiance [W/m^2]
+    dni_min                  float   Min. direct normal irradiance [W/m^2]
+    dni_max                  float   Max. direct normal irradiance [W/m^2]
+    dhi                      float   Mean diffuse horizontal irradiance [W/m^2]
+    dhi_std                  float   Std. diffuse horizontal irradiance [W/m^2]
+    dhi_min                  float   Min. diffuse horizontal irradiance [W/m^2]
+    dhi_max                  float   Max. diffuse horizontal irradiance [W/m^2]
+    lwd                      float   Mean. downward long-wave radiation [W/m^2]
+    lwd_std                  float   Std. downward long-wave radiation [W/m^2]
+    lwd_min                  float   Min. downward long-wave radiation [W/m^2]
+    lwd_max                  float   Max. downward long-wave radiation [W/m^2]
+    air_temperature          float   Air temperature [°C]
+    relative_humidity        float   Relative humidity [%]
+    pressure                 float   Atmospheric pressure [hPa]
+    =======================  ======  ==========================================
+
+    References
+    ----------
+    .. [1] `World Radiation Monitoring Center - Baseline Surface Radiation
+        Network (BSRN) `BSRN homepage <https:/https://bsrn.awi.de/>`_
+    .. [2] `Update of the Technical Plan for BSRN Data Management, 2013,
+       Global Climate Observing System (GCOS) GCOS-172.
+       <https://bsrn.awi.de/fileadmin/user_upload/bsrn.awi.de/Publications/gcos-174.pdf>`_
+    .. [3] `BSRN Data Retrieval via FTP
+       <https://bsrn.awi.de/data/data-retrieval-via-ftp/>`_
+    .. [4] `BSRN Data Release Guidelines
+       <https://bsrn.awi.de/data/conditions-of-data-release/>`_
+    """
+
+    # Read file and store the starting line number for each logical record (LR)
+    line_no_dict = {}
+    if str(filename).endswith('.gz'):  # check if file is a gzipped (.gz) file
+        with gzip.open(filename, 'rt') as f:
+            for num, line in enumerate(f):
+                if num == 1:  # Get month and year from the 2nd line
+                    start_date = pd.Timestamp(year=int(line[7:11]),
+                                              month=int(line[3:6]), day=1)

```

Comment:
Should the `Timestamp` include `tz='UTC'`?

Reply:
Yes, this is indeed a much nicer solution than the try statement and localizing later